bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BVM.java

```diff
@@ -3674,6 +3674,19 @@ private static boolean isAnydata(Collection<BType> types, Set<BType> unresolvedT
         return types.stream().allMatch(bType -> isAnydata(bType, unresolvedTypes));
     }
 
+    private static boolean isPureType(BType type, Set<BType> unresolvedTypes) {
+        if (type.getTag() == TypeTags.UNION_TAG) {
+            return ((BUnionType) type).getMemberTypes().stream()
+                    .allMatch(memType -> isPureType(memType, unresolvedTypes));
+        }
+
+        return isAnydata(type, unresolvedTypes) || type.getTag() == TypeTags.ERROR_TAG;
```

Comment:
Improvement: By swapping two conditions can improve the error case.

Reply:
Fixed with fbf9f4d.